train.py

```diff
@@ -112,16 +118,30 @@ def main(job_config: JobConfig):
     # 3. max_seq_len base on inputs
     model_config.norm_type = job_config.model.norm_type
     model_config.vocab_size = tokenizer.n_words
+    model_config.vocab_size = 50000
     model_config.max_seq_len = job_config.training.seq_len
 
     logger.info(f"Building {model_name} {job_config.model.flavor} with {model_config}")
-    with torch.device("meta"):
```

Comment:
Is there a reason we remove the meta device instantiation?

Reply:
i think it may theoretically be possible to use meta device and still load the weights (not init them), it is just not very clear how, not much documentation is available for the FSDP.

Reply:
for instance this [example](https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html) does not use the meta device

Reply:
If it doesn't break anything, I think we should keep it. There should be some benefits to speed for this. If it is breaking the transformers initialization, we can take it out for now.

Reply:
basically, you get an error when you load the weights using a state dict

Reply:
i am not sure it is fully supported to load the weights after sharding the model in this case, https://github.com/pytorch/pytorch/issues/114299

Reply:
Ok if it's throwing an error let's take it out.